File: problems/797/solution.go
package main
func allPathsSourceTarget(graph [][]int) [][]int {
    ans := [][]int{}
    
    dfs(graph, []int{0}, len(graph) - 1, &ans)
    
    return ans
}

func dfs(graph [][]int, path []int, target int, ans *[][]int) {
    last := path[len(path) - 1]
    
    if last == target {
        *ans = append(*ans, path)
        return
    }
    
    for _, next := range graph[last] {
        newPath := make([]int, len(path) + 1)
        copy(newPath, path)
        newPath[len(newPath) - 1] = next
        
        dfs(graph, newPath, target, ans)
    }
}

File: problems/797/solution.py
class Solution:
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        ans, q, target = [], deque([[0]]), len(graph) - 1
        
        while q:
            cur_path = q.popleft()
            
            if cur_path[-1] == target:
                ans.append(cur_path)
                continue

            for child in graph[cur_path[-1]]:
                q.append(cur_path + [child])
        
        return ans